parameter/ParameterMgr.cpp

```diff
@@ -2333,14 +2326,41 @@ bool CParameterMgr::exportDomainsXml(string& strXmlDest, bool bWithSettings, boo
 
     xmlDomainExportContext.setOutputRawFormat(_bOutputRawFormatIsHex);
 
-
-    return serializeElement(strXmlDest, xmlDomainExportContext, bToFile,
-                                    *pConfigurableDomains, strError);
+    return serializeElement(xmlDest, xmlDomainExportContext, *pConfigurableDomains, strError);
 }
 
 bool CParameterMgr::exportSingleDomainXml(string& strXmlDest, const string& strDomainName,
                                           bool bWithSettings, bool bToFile, string& strError) const
 {
+    CAutoLog autoLog(this, "Exporting single domain '" + strDomainName + "' to " +
+            (bToFile ? strXmlDest : "user-provided string"));
+
+    std::auto_ptr<ostream> output(NULL);
+
+    if (bToFile) {
+        output.reset(new ofstream(strXmlDest.c_str()));
+    } else {
+        output.reset(new ostringstream);
+        /* ostringstream* oss(new ostringstream); */
+        /* oss->rdbuf(strXmlDest.rdbuf()); */
+        /* output.reset(oss); */
+    }
+
+    // Do the actual export
+    if (!exportSingleDomainXml(*output, strDomainName, bWithSettings, strError)) {
+        return false;
+    }
+
+    if (!bToFile) {
+        strXmlDest = static_cast<ostringstream*>(output.get())->str();
```

Comment:
COPY PASTE ! The whole function is nearly the same as CParameterMgr::exportDomainsXml